### PR TITLE
Allow passing I18n locale to ActionMailer::Base#mail

### PR DESCRIPTION
**Pull request by c960657**
Based on (a little) positive feedback on my [post on rubyonrails-core](https://discuss.rubyonrails.org/t/feature-request-action-mailer-add-locale-support/75615), this PR adds better I18n locale support to Action Mailer.

Instead of doing this:
```
def welcome
  I18n.with_locale(@user.locale) do
    mail to: @user.email
  end
end
```

this PR allows this simpler syntax:
```
def welcome
  mail to: @user.email, locale: @user.locale
end
```

### CODE REVIEW

**Comment by c960657** [comment]
> This pull request has been automatically marked as stale because it has not had recent activity.

# 😞

File: actionmailer/lib/action_mailer/base.rb [hunk 1]
```diff
@@ -888,9 +894,12 @@ def set_content_type(m, user_content_type, class_default) # :doc:
       # If it does not find a translation for the +subject+ under the specified scope it will default to a
       # humanized version of the <tt>action_name</tt>.
       # If the subject has interpolations, you can pass them through the +interpolations+ parameter.
+      # The locale may be passing in the <tt>locale</tt> key, otherwise the mailer class default or
+      # `I18n.locale` will be used.
       def default_i18n_subject(interpolations = {}) # :doc:
         mailer_scope = self.class.mailer_name.tr("/", ".")
-        I18n.t(:subject, **interpolations.merge(scope: [mailer_scope, action_name], default: action_name.humanize))
+        locale = interpolations.key?(:locale) ? interpolations[:locale] : compute_default(self.class.default[:locale])
+        I18n.t(:subject, **interpolations.merge(scope: [mailer_scope, action_name], default: action_name.humanize, locale: locale))
```
Comment (rafaelfranca) [inline actionmailer/lib/action_mailer/base.rb:902]: If `interpolations.key?(:locale)` is true would not this method call have two `locale` options with the later overriding the former argument? I think we should only pass this option if `interpolations.key?(:locale)` is false.

File: actionmailer/CHANGELOG.md [hunk 2]
```diff
@@ -1,4 +1,8 @@
+## Unreleased
 
+*   Add `locale` argument for `ActionMailer::Base#mail`.
```
Comment (rafaelfranca) [inline actionmailer/CHANGELOG.md:3]:
```suggestion
*   Add `locale` argument for `ActionMailer::Base#mail`.
```

**Review by rafaelfranca — COMMENTED** [review]
Just two small comments but the changes look good. Thank you for the pull request and sorry for the delay.

File: actionmailer/lib/action_mailer/base.rb (hunk 1 above)
Comment (c960657) [reply actionmailer/lib/action_mailer/base.rb:902]: I'm not sure I understand your question. Is this a comment on style or functionality? 

The following is functionally equivalent. Do you prefer this approach?
```
        options = interpolations.key?(:locale) ? {} : { locale: compute_default(self.class.default[:locale]) }
        I18n.t(:subject, **interpolations.merge(scope: [mailer_scope, action_name], default: action_name.humanize), **options)
```

Comment (c960657) [reply actionmailer/lib/action_mailer/base.rb:902]: @rafaelfranca Ping :-)

Comment (c960657) [reply actionmailer/lib/action_mailer/base.rb:904]: @rafaelfranca Ping :-)

**Pull request closed without merging**